srai/loaders/osm_loaders/osm_tile_loader.py

```diff
@@ -0,0 +1,127 @@
+"""
+OSM tile loader.
+
+This module implements downloading tiles from given OSM tile server.
+"""
+from io import BytesIO
+from pathlib import Path
+from typing import Any, Callable, Optional, Union
+from urllib.parse import urljoin
+
+import pandas as pd
+import requests
+from PIL import Image
+
+from srai.regionizers.slippy_map_regionizer import SlippyMapRegionizer
+from srai.utils import geocode_to_region_gdf
+
+from .osm_tile_data_collector import DataCollector, InMemoryDataCollector, SavingDataCollector
+
+
+class TileLoader:
+    """
+    Tile Loader.
+
+    Downloads raster tiles from user specified tile server, like listed in [1]. Loader founds x, y
+    coordinates [2] for specified area and downloads tiles. Address is built with schema
+    {tile_server_url}/{zoom}/{x}/{y}.{resource_type}
+
+    References:
+        1. https://wiki.openstreetmap.org/wiki/Raster_tile_providers
+        2. https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames
+    """
+
+    def __init__(
+        self,
+        tile_server_url: str,
+        zoom: int,
+        verbose: bool = False,
+        resource_type: str = "png",
+        auth_token: Optional[str] = None,
+        collector_factory: Optional[Union[str, Callable[[], DataCollector]]] = None,
+        storage_path: Optional[Union[str, Path]] = None,
+    ) -> None:
+        """
+        Initialize TileLoader.
+
+        Args:
+            tile_server_url (str): url of tile server, without z, x, y parameters
+            zoom (int): zoom level [1]
+            verbose (bool, optional): should print logs. Defaults to False.
+            resource_type (str, optional): file extension. Added to the end of url.
+                Defaults to "png".
+            auth_token (Optional[str], optional): auth token. Added as access_token parameter
+                to request. Defaults to None.
+            collector_factory (Optional[Union[str, Callable[[], DataCollector]]], optional):Function
+                returning DataCollector. If None uses InMemoryDataCollector. Defaults to None.
+                    If `return` uses  InMemoryDataCollector
+                    If `save` uses  SavingDataCollector
```

Comment:
```suggestion
            collector_factory (Optional[Union[str, Callable[[], DataCollector]]], optional): Function
                returning DataCollector. If None uses InMemoryDataCollector. Defaults to None.
                If `return` uses  InMemoryDataCollector
                If `save` uses  SavingDataCollector
```

Reply:
Also, maybe consider using enum? I don't like magic strings (they should be acceptable - look at OSMWayLoader and NetworkType)